sensor_harm/harmonization_model.py

```diff
@@ -399,13 +422,21 @@ def process_NBAR(img_dir, scene_id: str, bands, sz_path, sa_path, vz_path, va_pa
         logging.debug(list(filter(r.match, imgs_in_dir)))
 
         # TODO: We should use file name. Check which of them have same filename and try to get from some angle band
-        if scene_id.startswith('S2'):
+        if is_sentinel2(scene_id):
             input_file = Path(list(filter(r.match, imgs_in_dir))[0])
-        else:
-            _entry = list(Path(img_dir).glob(f'{scene_id}_{b}.TIF'))[0]
-            input_file = Path(_entry.name)
+            output_file = out_dir.joinpath(Path(input_file).stem + '_NBAR').with_suffix('.tif')
+        elif is_landsat(scene_id):
+            if dataset_collection == '01':
```

Comment:
Why you need another function argument `dataset_collection` since the `scene_id` has the info related Collection Level?

Reply:
I agree! You may be using scene_id parser, just like the one we use in bdc-tiler, in this function you can get all image parameters. See in: https://github.com/brazil-data-cube/bdc-tiler/blob/master/bdc_tiler/utils/scene_parser.py#L48